docs/typescript/programming-guide.md

```diff
@@ -1,331 +1,266 @@
+      },
+      body: JSON.stringify({ name })
+    });
+    console.log(`>>> STEP 1: Signed the guestbook for ${name}`);
+  }
 
-export class Greetings {
-  //Omitted for brevity: @GetApi('/') //app readme
+  // Create a database connection using Knex.js
+  static db = knex(knexConfig);
 
-  @Transaction()
-  static async InsertGreeting(ctxt: TransactionContext<Knex>, gr: GreetingRecord) {
-    await ctxt.client('greetings').insert(gr);
-    ctxt.logger.info(`Greeting to ${gr.name} recorded in the database!`);
+  // Record the greeting in the database using Knex.js
+  static async insertGreeting(name: string): Promise<void> {
+    await Guestbook.db('dbos_greetings').insert({ greeting_name: name });
+    console.log(`>>> STEP 2: Greeting to ${name} recorded in the database!`);
   }
 
-  @GetApi('/greeting/:friend')
-  static async Greeting(ctxt: HandlerContext, friend: string) {
-    const noteContent = `Thank you for being awesome, ${friend}!`;
-    await ctxt.invoke(Greetings).InsertGreeting(
-      { name: friend, note: noteContent }
-    );
-    return noteContent;
+  static async greetingEndpoint(name: string): Promise<string> {
+    await Guestbook.signGuestbook(name);
+    await Guestbook.insertGreeting(name);
+    return `Thank you for being awesome, ${name}!`;
   }
 }
-```
 
-Here we define a `GreetingRecord` interface matching a row of data in our `greetings` table. We then define a `@Transaction` called `InsertGreeting` that inserts a new `GreetingRecord` into `greetings`. Finally, we add a line to the GET API function `Greeting` to invoke `InsertGreeting` with the provided `name` and a welcoming `note`.
+// Create an HTTP server using Express.js
+export const app = express();
+app.use(express.json());
 
-:::info
-In this quickstart, we run queries using the [Knex query builder](https://knexjs.org/guide/query-builder.html). DBOS Transact also supports [Drizzle](../typescript/tutorials/using-drizzle.md), [TypeORM](../typescript/tutorials/using-typeorm), [Prisma](../typescript/tutorials/using-prisma), and [Raw SQL](../typescript/tutorials/transaction-tutorial).
-:::
+app.get('/greeting/:name', async (req: Request, res: Response): Promise<void> => {
+  const { name } = req.params;
+  res.send(await Guestbook.greetingEndpoint(name));
+});
 
-Stop your app with CTRL+C. Rebuild with `npm run build` and start with `npx dbos start`. Make a few visits to the greeting URL in your browser, i.e. http://localhost:3000/greeting/Mike. With every new visit, the app should print this to the console:
-```
-[info]: Greeting to Mike recorded in the database! 
+async function main() {
+  const PORT = 3000;
+  const ENV = process.env.NODE_ENV || 'development';
+
+  app.listen(PORT, () => {
+    console.log(`🚀 Server is running on http://localhost:${PORT}`);
+    console.log(`🌟 Environment: ${ENV}`);
+  });
+}
+
+main().catch(console.log);
 ```
 
-### 3.1. Reading from the Database
+Build your app with `npm run build` and start it with `npx dbos start`.
```

Comment:
So we can also use nodemon to continuously pull latest changes in code. `npm run dev` works.